packages/graph/hash_graph/lib/graph/src/types/uri.rs

```diff
@@ -0,0 +1,81 @@
+use std::{fmt, result::Result as StdResult, str::FromStr};
+
+use error_stack::{Context, IntoReport, Report, Result, ResultExt};
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+pub type BaseId = String;
+pub type BaseIdRef = str;
```

Comment:
Newtypes (and aliases) for strings suck 🙃 

How do we feel about `BaseUri(String)` with a deref for once, does that solve the whole `BaseUriRef` thing?